packages/fast-jss-manager-react/src/design-system-provider.tsx

```diff
@@ -6,36 +6,31 @@
  */
 import * as React from "react";
 import * as propTypes from "prop-types";
+import { Consumer, Provider } from "./context";
 
+export type IDesignSystem<T> = T extends {[key: string]: unknown} ? T : never;
 /**
  * Describes the props that the DesignSystemProvider uses. It accepts a single prop "designSystem"
  * that gets exposed to all downstream components of the DesignSystemProvider
  */
 export interface IDesignSystemProviderProps<T> {
-    designSystem?: T;
+    designSystem: IDesignSystem<T>;
 }
 
-class DesignSystemProvider<T> extends React.Component<IDesignSystemProviderProps<T>, {}> {
-    // TODO: How can we type these to be generics but also static properties
-    public static contextTypes: any = {
-        designSystem: propTypes.any
-    };
-
-    public static childContextTypes: any = {
-        designSystem: propTypes.any
-    };
-
-    public getChildContext(): IDesignSystemProviderProps<T> {
-        const designSystem: T = this.props.designSystem;
-        const contextualDesignSystem: T = this.context.designSystem;
-        const context: T = Object.assign({}, contextualDesignSystem, designSystem);
-
-        return { designSystem: context };
+export class DesignSystemProvider<T> extends React.Component<IDesignSystemProviderProps<T>, {}> {
+    public render(): React.ReactNode {
+        return (
+            <Consumer>
```

Comment:
Can you help me understand this "outer" `<Consumer />`? 

I'm probably missing something, I'm only aware of the pattern from the docs of
```
<Provider>
    <Consumer>
</Provider>
```

Is our React output more similar to below?
```
<Consumer>
    <Provider>
        <Consumer>
               // Child / Children
        </Consumer>
    </Provider>
</Consumer>
```

Reply:
The outer consumer is to consume the parent context. As far as I can tell, there is no way for a provider to consume context context values, so in order to inherit context values we need to first render a consumer, which *then* gives the provider the parent context to merge with its own props.